### DISCUSSION TITLE
[Feat]: ⭐ Roadmap

### DISCUSSION THREAD

**Discussion by davidjerleke** [davidjerleke/embla-carousel#841] [Ideas]
#  ⭐ Roadmap

### `embla-carousel`
- [x] Remove the `init` event (deprecated).
- [x] #202.
- [x] #1078.
- [x] #1217
- [x] #1043.
- [x] #1043 Rename `scrollTo --> scrollToSnap` .
- [x] #1087.
- [x] #1091.
- [ ] #738.

### `embla-carousel-autoplay`
- [x] #743.
- [x] #1059.
- [x] #1175.
- [x] #1021.
- [ ] #821.

### `embla-carousel-auto-height`
- [x] #1216
- [x] #1215.

### `embla-carousel-class-names`
- [x] #1060.  
- [x] New **option** `isPointerDown` https://github.com/davidjerleke/embla-carousel/discussions/834#discussioncomment-9221995:

```tsx
export const defaultOptions: OptionsType = {
  active: true,
  breakpoints: {},
  snapped: 'is-snapped',
  inView: 'is-in-view',
  pointerDown: 'is-pointer-down', // Applied while pointer is down on the carousel. (Previously is-dragging)
  dragging: 'is-dragging', // Applied while the carousel is being dragged. (New behaviour)
  draggable: 'is-draggable',
}
```

### `embla-carousel-svelte`
- [x] Rename the event to lowercase `onemblaInit` --> `onemblainitt`.
- [ ] #1213.

### `embla-carousel-docs`
- [ ] #702.
- [x] #1218.

### `embla-carousel-data-attributes`
- [ ] #1051 (build a new plugin).

**Comment by sarussss** [comment]
> Remove the init event (deprecated).

Hi @davidjerleke.
I'm really confused, when I read this because I'm using that event, if this event is removed, is there any other event that can do this? Is the removal of this event related to the development of SSR?

**Reply by davidjerleke** [reply]
@sarussss what Embla package are you using? No it’s not related to SSR.

**Reply by sarussss** [reply]
@davidjerleke i'm using Vanilla module package v8.0.1 
[Is this the event mentioned?](https://www.embla-carousel.com/api/events/#init)

**Reply by davidjerleke** [reply]
> [Is this the event mentioned?](https://www.embla-carousel.com/api/events/#init)

Yes it is.

> @davidjerleke i'm using Vanilla module package v8.0.1 

Allright. This is how you should do it instead:

```tsx
import EmblaCarousel from 'embla-carousel'

function onInit(emblaApi) {
  /* Do something on init... */ 
}

const emblaNode = document.querySelector('.embla')
const emblaApi = EmblaCarousel(emblaNode, { loop: true })

onInit(emblaApi) // Embla is ready here
emblaApi.on('reInit', onInit)
```

**Reply by sarussss** [reply]
@davidjerleke Thank you.

**Reply by sarussss** [reply]
Hi @davidjerleke 
What do you think if you add it to the roadmap?
I think the library would be even better if it could improve this problem.
https://github.com/davidjerleke/embla-carousel/discussions/676

**Comment by edtorba** [comment]
Hello 👀 Have you thought about creating a package using the format e.g. `@embla-carousel`, `@embla-carousel/autoplay`, etc.? It could make dependency management easier, improve discoverability, and keep everything more organised under a consistent naming structure. Just a thought! Thanks! 😄
